### PR TITLE
Merge release-v2.0 to main (as of v2.0.0-beta.15)

### PR DESCRIPTION
**Pull request by 1ec5**
This PR merges the release-v2.0 branch to the main branch. The release-v2.0 branch is where we’ve been developing MapboxNavigation v2.0.0 over the past six months or so. A release-v1.4 branch has already been created with the current state of the main branch, in order to track any out-of-band patch releases that may be required over the coming months. However, v2._x_ has already been the focus of development for some time. Synchronizing the main branch ensures that nothing slips through the cracks as we put out v2.0.0 and sets the stage for v2.1 work to begin soon.

v2.0.0 is shaping up to be a large release, so this PR isn’t very easy to read closely. [The changelog](https://github.com/mapbox/mapbox-navigation-ios/blob/release-v2.0/CHANGELOG.md#v200) is quite lengthy, but we also have [a comprehensive migration guide](https://github.com/mapbox/mapbox-navigation-ios/wiki/Upgrading-from-v1.x-to-v2.0.x) for anyone who hasn’t been following the betas and needs to catch up. Of particular note are the discontinuation of full Carthage support in favor of full Swift Package Manager support, the discontinuation of iOS 10 compatibility, and the migration to a new license, which is detailed in #3115.

/cc @mapbox/navigation-ios

### CODE REVIEW

**Comment by 1ec5** [comment]
I’ve temporarily disabled all the requirements for passing builds on PRs targeting the main branch, since this PR would replace a number of them with distinct builds, which we can enable after merging.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -112,17 +118,38 @@ The .strings files should still be in the original English – that’s expecte
 
 ## Adding tests
 
+### Supported devices:
+- iPhone 11 Pro Max, iOS 13.7
+- iPhone 12 Pro Max, iOS 14
```
Comment (S2Ler) [inline CONTRIBUTING.md:123]:
```suggestion
- iPhone 12 Pro Max, iOS 14.4
```

**Review by S2Ler — APPROVED** [review]
🚀 Congrats to the team, huge effort!

**Comment by MaximAlien** [comment]
I was wondering why I do not see all commits. Turns out GitHub shows only 250 last commits.

**Review by MaximAlien — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,115 @@
...
+
+### Electronic horizon
+
+* While a `RouteController`, `PassiveLocationDataSource`, or `PassiveLocationManager` is tracking the user’s location, you can get notifications about location changes that indicate relevant details in the _electronic horizon_ – the upcoming portion of the routing graph – such as the names of cross streets and upcoming speed limit changes. To receive this information, observe the `Notification.Name.electronicHorizonDidUpdatePosition`, `Notification.Name.electronicHorizonDidEnterRoadObject`, `Notification.Name.electronicHorizonDidExitRoadObject`, and `Notification.Name.electronicHorizonDidPassRoadObject` notifications. Use the `RouteController.roadGraph` or `PassiveLocationDataSource.roadGraph` property to get more information about the edges contained in these notifications. Optionally set the `RouteController.electronicHorizonOptions` or `PassiveLocationDataSource.electronicHorizonOptions` property to configure the extent of electronic horizon calculations. ([#2834](https://github.com/mapbox/mapbox-navigation-ios/pull/2834))
```
Comment (MaximAlien) [inline CHANGELOG.md:56]: We should probably fully refine changelog before going GA. Some items are no longer accurate (e.g. `PassiveLocationDataSource` mentioned in severla places, but no longer present).

File: .circleci/config.yml [hunk 3]
```diff
@@ -261,78 +254,86 @@ jobs:
...
+          device: "iPhone 11 Pro Max"
+      - spm-test-job:
+          name: "swift test; Xcode 12.4; iOS 14.4"
+          xcode: "12.4.0"
+          iOS: "14.4"          
+          device: "iPhone 12 Pro Max"
       - ios-trigger-metrics:
           filters:
             branches:
-              only: main
```
Comment (azarovalex) [inline .circleci/config.yml:338]: Looks like we should change it back to main.

File: README.md [hunk 4]
```diff
@@ -67,22 +88,19 @@ To install Mapbox Navigation using [Carthage](https://github.com/Carthage/Cartha
...
+1. Run `carthage bootstrap --platform iOS --use-xcframeworks --cache-builds --use-netrc`.
 
-### Using Swift Package Manager
+1. Follow the rest of [Carthage’s iOS integration instructions](https://github.com/Carthage/Carthage#if-youre-building-for-ios-tvos-or-watchos). Your application target’s Embed Frameworks build phase should include `MapboxCoreNavigation.xcframework`, `MapboxNavigationNative.xcframework`, and `MapboxCommon.xcframework`.
 
-The MapboxNavigation framework [does not yet support Swift Package Manager](https://github.com/mapbox/mapbox-navigation-ios/issues/2629) either on the command line or in Xcode. However, if you are building a user interface from scratch, you can [install just the MapboxCoreNavigation framework using SPM](./custom-navigation.md#using-swift-package-manager). 
+MapboxNavigation no longer supports Carthage as of v2.0.0-alpha.1. However, if you are building a user interface from scratch, you can [install just the MapboxCoreNavigation framework using Carthage](./custom-navigation.md#using-carthage).
 
 ## Configuration
```
[the remark this answers is not in the document]
Comment (1ec5) [reply README.md:105]: Interesting, I thought we had finally massaged the readme in both branches to be accurate with respect to what was supported in either branch. The outdated language must’ve come in as part of a merge from main at some point.

File: CONTRIBUTING.md (hunk 1 above)
Comment (1ec5) [reply CONTRIBUTING.md:129]: Fixed in 1e1b41e320244d620ded7c1b8e81d03a4923a8e4 on the main branch – sorry for missing this feedback earlier.

**Pull request merged**